app/main.py

```diff
@@ -1,3 +1,47 @@
-def shop_trip():
-    # write your code here
-    pass
+import json
+
+from app.customer import Customer
+from app.shop import Shop
+
+
+def shop_trip() -> None:
+    with open("app/config.json", "r") as file:
+        file_config = json.load(file)
+
+    for customer in file_config["customers"]:
+        buyer = Customer(
+            customer["name"],
+            customer["product_cart"],
+            customer["location"],
+            customer["money"],
+            customer["car"]["fuel_consumption"]
+        )
+
+        buyer.print_customer_name_and_money()
+
+        shops_and_prices = {}
+        instance_shops = []
+        for shop in file_config["shops"]:
+            shop = Shop(shop["name"], shop["location"], shop["products"])
+            price = buyer.get_price_product_cart(shop, file_config)
+            shops_and_prices[shop.name] = price
+            instance_shops.append({shop.name: shop})
+        name_shop_whit_best_price = min(
```

Comment:
There's a typo in the variable name `name_shop_whit_best_price`. It should be `name_shop_with_best_price`.